src/buffer/out/textBuffer.cpp

```diff
@@ -1524,9 +1524,15 @@ til::point TextBuffer::GetGlyphEnd(const til::point pos, bool accessibilityMode,
 bool TextBuffer::MoveToNextGlyph(til::point& pos, bool allowExclusiveEnd, std::optional<til::point> limitOptional) const
 {
     const auto bufferSize = GetSize();
-    const auto limit{ limitOptional.value_or(bufferSize.EndExclusive()) };
-
-    const auto distanceToLimit{ bufferSize.CompareInBounds(pos, limit, true) };
+    bool pastEndInclusive = false;
+    const auto limit = [bufferSize, limitOptional](bool& pastEndInclusive) {
+        const auto endInclusive{ bufferSize.BottomRightInclusive() };
+        const auto val = limitOptional.value_or(endInclusive);
+        pastEndInclusive = val > endInclusive;
+        return pastEndInclusive ? endInclusive : val;
+    }(pastEndInclusive);

```

Comment:
Hmm I thought I had commented it, but it seems like I forgot...

I more standard way to write this would be this IMHO:
```cpp
bool pastEndInclusive;
til::point limit;
{
    const auto endInclusive = bufferSize.BottomRightInclusive();
    const auto val = limitOptional.value_or(endInclusive);
    pastEndInclusive = val > endInclusive;
    limit = pastEndInclusive ? endInclusive : val;
}
```

I believe that would be easier to parse but also to debug (when you use F10 stepwise debugging it won't "jump around" due to a function call).

Another option is:
```cpp
const auto endInclusive = bufferSize.BottomRightInclusive();
auto limit = limitOptional.value_or(endInclusive);
const auto pastEndInclusive = limit > endInclusive;

if (pastEndInclusive)
{
    limit = endInclusive;
}
```

Finally, you can also just write `[&]` and drop the lambda parameter and explicit arguments.

Reply:
added a comment too